src/main/java/bobby/task/Deadline.java

```diff
@@ -0,0 +1,37 @@
+package bobby.task;
+
+public class Deadline extends Task{
+    private String by;
+
+    public Deadline(String fullTaskDescription) {
+        super(fullTaskDescription);
+        this.by = getBy();
+    }
+
+    public String getType() {
+        return "Deadline";
+    }
+
+    public String getBy() {
+        String[] descriptionWords = fullTaskDescription.split(" /by ", 2);
+        String by = descriptionWords[1].trim();
+        return by;
+    }
+
+    // get formatted description to print out for list and task commands, eg "return book (by: Sunday)"
+    public String getFormattedDescription() {
+        return String.format("%s (by: %s)", getTask(),this.by);
+    }
+
+    public String getFormattedFileDescription() {
+        return String.format("Deadline,%s,%s,%s",getStatusIcon(), getTask(), this.by);
```

Comment:
The number of spaces here between each argument of the method is not consistent.